app/locations/controllers.js

```diff
@@ -1,10 +1,27 @@
-const { sortBy } = require('lodash')
+const { get, sortBy } = require('lodash')
 
-function locations(req, res) {
-  const locations = sortBy(req.userLocations, 'title')
+const referenceDataService = require('../../common/services/reference-data')
+
+async function locations(req, res, next) {
+  const userPermissions = get(req.session, 'user.permissions', [])
+
+  let locations = []
+  let regions = []
+
+  if (userPermissions.includes('allocation:create')) {
+    try {
+      regions = await referenceDataService.getRegions()
+    } catch (error) {
+      next(new Error('Failed to retrieve the regions'))
+      return
+    }
+  } else {
+    locations = sortBy(req.userLocations, 'title')
```

Comment:
Could you use this as the default value for `locations` instead of having to have an extra `else`?

Reply:
Ah, I think this else is what is still causing a user with both roles to not be able to see locations.

I feel like we should always pass locations to the view and that regions should be behind a permission.

That way users still have the ability to see a particular location if they want but also the ability to see locations grouped by region.

Reply:
I don't know how the backend works, but they will be able to see all moves for a region I believe.
Is the API checking the permissions for a user?